building/windows.md

```diff
@@ -0,0 +1,240 @@
+
+```
+
+Adding toolchain binaries to the PATH variable also can be tricky. On Windows, we move over folders with names
+containing white spaces, after using:
+
+``` bash
+echo "export PATH=$PATH:$HOME/toolchains/i386-pc-phoenix/i386-pc-phoenix/bin/:$HOME/toolchains/arm-phoenix/arm-phoenix/bin/:$HOME/toolchains/riscv64-phoenix/riscv64-phoenix/bin/:$HOME/toolchains/sparc-phoenix/sparc-phoenix/bin/" >> $HOME/.bashrc
+```
+
+After calling this we need to fix paths in `.bashrc` file e.g. `C:\x\y z\d` we must change to `C:\x\'y z'\d`.
+
+This simple change will fix all unwanted messages at
+the start of the `Cygwin` terminal, and it will allow us to use the toolchain without setting the path every time.
+
+Now it should be possible to build the Phoenix-RTOS system image with the full configuration excluding tests:
+
+```bash
+TARGET=ia32-generic-qemu CONSOLE=serial phoenix-rtos-build/build.sh all
+```
+
+Of course, you can build the system image for this target in different configuration.
+More information [here](../building/script.md).
+
+In case of performing all steps right, the statement about successful build will be printed. The kernel and disk images
+will be available in the `_boot` directory then.
+
+Then you can follow instructions from
+[ia32-generic-qemu quick-start](../quickstart/ia32-generic-qemu.md).
+
+* Note that QEMU is already installed, so this step may be skipped.
+
+## WSL (Windows Subsystem for Linux)
+
+This part is only available for users with Windows 10 **(version 1709 or later)** and all versions of Windows 11
+
+`WSL` is based on a Linux virtualization environment within a Windows system.
+To properly install `WSL` it's needed to enable **virtualization** in BIOS settings as by default it's disabled.
+
+`WSL` we can download using `winget` in PowerShell (Remember to run it as an administrator):
+
+```powershell
+winget install wsl
+```
+
+If `winget` is not available on your system, search for the `App Installer` in the Microsoft Store.
+
+After that, we'll need to set up an environment. In our case, we're aiming for `Ubuntu-22.04`.
+To install it, use the following command:
+
+```powershell
+wsl --install -d Ubuntu-22.04
+```
+
+`WSL` will install the subsystem and all needed components. After successful installation, we can type
+
+```powershell
+wsl
+```
+
+In PowerShell and the entire subsystem will start up (This might take some time).
+On the first start, you will be prompted to create an account in the subsystem (sudo credentials).
+
+![Sudo Credentials](_images/wsl-sudo-credentials.png)
+
+After that simply follow the instructions written in the [Building Phoenix-RTOS](README.md) and
+[ia32-generic-qemu quick-start](../quickstart/ia32-generic-qemu.md).
+
+(Quick tip: It is not mandatory but turning off Real-Time Protection in Windows Defender speeds up the toolchain
+building process)
+
+### See also
+
+1. [Building Phoenix-RTOS image](README.md)

```

Comment:
I think that it would be also good to place there link to ia32-generic quickstart.